code/modules/research/ordnance/tank_compressor.dm

```diff
@@ -0,0 +1,330 @@
+#define TANK_COMPRESSOR_PRESSURE_LIMIT 5000
+#define TANK_COMPRESSOR_MAX_TRANSFER_RATE 20
+#define SIGNIFICANT_AMOUNT_OF_MOLES 10
+
+/obj/machinery/atmospherics/components/binary/tank_compressor
+	name = "Tank Compressor"
+	desc = "Heavy duty shielded air compressor designed to pressurize tanks above the safe limit."
+	circuit = /obj/item/circuitboard/machine/tank_compressor
+	icon = 'icons/obj/machines/research.dmi'
+	base_icon_state = "tank_compressor"
+	icon_state = "tank_compressor-open"
+	density = TRUE
+
+	var/active = FALSE
+	var/transfer_rate = TANK_COMPRESSOR_MAX_TRANSFER_RATE
+	var/datum/gas_mixture/leaked_gas_buffer
+	var/list/compressor_record
+	var/last_recorded_pressure = 0
+	var/record_number = 1
+	var/obj/item/tank/inserted_tank
+	/// Reference to a disk we are going to print to.
+	var/obj/item/computer_hardware/hard_drive/portable/inserted_disk
```

Comment:
Same question here, though it applies to the tank too.
On another note, how different is our inserted_disk from the doppler array's

Reply:
Also remember to null these on Destroy, deconstruct won't always happen

Reply:
1. Unlike the disks, tanks being deleted here will be caught by a signal and nullified properly since they are actually designed to be deleted inside the compressor,

2. Pretty much the same idea to be honest, whatever experiment an explosion is eligible for is determined by the doppler and the compressor. Doppler's is way more complicated though.

3. Used QDEL_NULL for destroy now, let me know if this is problematic.